cpg-core/src/main/java/de/fraunhofer/aisec/cpg/TranslationConfiguration.java

```diff
@@ -276,6 +286,11 @@ public Builder sourceLocations(List<File> sourceLocations) {
       return this;
     }
 
+    public Builder loadCompilationDatabase(Map<File, List<String>> compilationDatabase) {
```

Comment:
```suggestion
    public Builder useCompilationDatabase(Map<File, List<String>> compilationDatabase) {
```

Reply:
Wouldn't it make more sense to provide the File to the compilation database here and extract the necessary entries in the CXX language frontend?

Reply:
If we assume that there are more languages with this format of compilation databases of multiple implementations of a frontend for the same language I would not put it into the specific implementation. What do you guys think?

Reply:
For now, `loadCompilationDatabase(Map<File, List<String>> compilationDatabase)` is  changed to `useCompilationDatabase(File compilationDatabaseFile)` and the function that returns the map(`Map<File, List<String>>`) of compilation database is placed inside `cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CompilationDB.kt`. 
@konradweiss   maybe we can have different implementations for getting a compilation database  for different languages in their respective `/frontends/<language>/CompilationDB.kt`? What do you think about this?

Reply:
Yes that option would definitely speak for the approach to move it to the frontend itself like @oxisto proposed